### PR TITLE
bug(help): resolve issue when using --help

### LINKED ISSUE #3195: `ng help command` not supported while it works in ember-cli
> Please provide us with the following information:
> ---------------------------------------------------------------

macOS 10.12.1 (16B2555)

angular-cli: 1.0.0-beta.20-4
node: 6.9.1
os: darwin x64

Just run:
```
ng help set
```
You can substitute `set` for any other CLI command.

```
$ ng help set
undefined
```

### Mention any other details that might be useful.

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

Ember CLI allows to print help for a specific command:
```
$ ember help build
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
Requested ember-cli commands:

ember build <options...>
  Builds your app and places it into the output path (dist/ by default).
  aliases: b
  --environment (String) (Default: development)
    aliases: -e <value>, -dev (--environment=development), -prod (--environment=production)
  --output-path (Path) (Default: dist/)
    aliases: -o <value>
  --watch (Boolean) (Default: false)
    aliases: -w
  --watcher (String)
  --suppress-sizes (Boolean) (Default: false)
```

This is extremely useful as the whole help can be quite large and hard to navigate.

### PR DESCRIPTION
**Pull request by Brocco**
Fixes #3195 and #3168

### CODE REVIEW

File: packages/angular-cli/ember-cli/lib/cli/cli.js [hunk 1]
```diff
@@ -25,6 +25,16 @@ module.exports = CLI;
 CLI.prototype.run = function(environment) {
   return Promise.hash(environment).then(function(environment) {
     var args = environment.cliArgs.slice();
+
+    if (args[0] === '--help') {
+      if (args.length === 1) {
+        args[0] = 'help';
+      } else {
+        args.shift();
+        args.push('--help');
```
Comment (hansl) [inline packages/angular-cli/ember-cli/lib/cli/cli.js]: Push back the same value?

Comment (Brocco) [inline packages/angular-cli/ember-cli/lib/cli/cli.js]: Yes, --help is an option, not a command. So remove it from the start of the array and push it to the end.

**Comment by Brocco** [comment]
e2e test added

File: packages/angular-cli/ember-cli/lib/cli/cli.js (hunk 1 above)
Comment (hansl) [inline packages/angular-cli/ember-cli/lib/cli/cli.js]: Nevermind, I was reading `unshift`.  Cheers!

File: tests/e2e/tests/commands/help/help-option.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import {silentNg} from '../../../utils/process';
+
+
+export default function() {
+  return Promise.resolve()
+    .then(() => silentNg('--help'))
+    .then(() => process.chdir('/'))
+    .then(() => silentNg('--help'));
```
Comment (hansl) [inline tests/e2e/tests/commands/help/help-option.ts]: Mind trying with a command as well?

**Pull request merged**